packages/web-components/src/components/va-radio/va-radio.tsx

```diff
@@ -74,6 +74,13 @@ export class VaRadio {
    * Insert a header with defined level inside the label (legend)
    */
   @Prop() labelHeaderLevel?: string;
+
+  /**
+   * An optional message that will be read by screen readers when the header is focused. The label-header-level
+   * prop must be set for this to be active.
+   */
+  @Prop() headerAriaDescribedby?: string;
```

Comment:
I've made this specific to the (optional) header but I had been thinking about **what if** this is wanted or useful when that isn't set. Just the default `legend` text. 

That opened up a can of worms for me because it seemed like a header tag within a legend behaved in different ways when trying to connect it with additional context (using aria-describedby). It seemed like I would instead need to use `aria-labelledby` on the `fieldset` (this doesn't work with the header tag though).

I also didn't want to try and solve for something that wasn't actually needed or wanted so I'm adding this as something that is _only_ available with `label-header-level` (which was the original ask).

**Markup With header level prop**

```html
<fieldset>
  <legend>
    <h1>My label text</h1>
  </legend>
</fieldset>
```

**Markup Without header prop**

```html
<fieldset>
  <legend>
    My label text
  </legend>
</fieldset>
```

Reply:
If we do want that additional screen reader text for both header and non-header text, an alternate approach could be to do something like append the header text and the legend text with the additional string (setting `sr-only` on it) instead of using any `aria-` variation. Something like:

```js
{HeaderLevel ? (
  <HeaderLevel part="header">
    {label} <span class="sr-only">{headerAriaDescribedby}</span>
  </HeaderLevel>
) : (
  {label} <span class="sr-only">{headerAriaDescribedby}</span>
)}
```

That's what this article is saying was their solution:

https://blog.tenon.io/accessible-validation-of-checkbox-and-radiobutton-groups/